lib/SPIRV/SPIRVReader.cpp

```diff
@@ -4364,6 +4364,51 @@ bool llvm::getSpecConstInfo(std::istream &IS,
   return !IS.bad();
 }
 
+bool llvm::readSPIRVHeader(std::istream &IS, SPIRVHeaderData &Result,
+                           std::string &ErrMsg) {
+  std::unique_ptr<SPIRVModule> BM(SPIRVModule::createSPIRVModule());
+  BM->setAutoAddExtensions(false);
+  SPIRVDecoder D(IS, *BM);
+  SPIRVWord Magic;
+  SPIRVWord Buffer;
+  bool Success;
+  D >> Magic;
+  if (!BM->getErrorLog().checkError(Magic == MagicNumber, SPIRVEC_InvalidModule,
+                                    "invalid magic number")) {
+    ErrMsg = "invalid magic number";
+    return false;
+  }
+
+  D >> Buffer;
+  Result.Version = Buffer;
```

Comment:
Why not `D >> Result.Version` ?

Reply:
Actually I just wanted to assign values to struct fields in the same way using variable `Buffer`. I will change it after I get another requested changes just not to spam with commits.